lms/lib/courseware_search/lms_filter_generator.py

```diff
@@ -3,13 +3,37 @@
     * Filter by all courses in which the user is enrolled in
 """
 from student.models import CourseEnrollment
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
 
 from search.filter_generator import SearchFilterGenerator
+from openedx.core.djangoapps.course_groups.cohorts import get_cohort
+from openedx.core.djangoapps.course_groups.models import CourseUserGroupPartitionGroup
 
 
 class LmsSearchFilterGenerator(SearchFilterGenerator):
     """ SearchFilterGenerator for LMS Search """
 
+    def filter_dictionary(self, **kwargs):
+        """ base implementation which filters via start_date """
+        filter_dictionary = super(LmsSearchFilterGenerator, self).filter_dictionary(**kwargs)
+        if 'user' in kwargs and 'course_id' in kwargs and kwargs['course_id']:
+            try:
+                course_key = CourseKey.from_string(kwargs['course_id'])
+            except InvalidKeyError:
+                course_key = SlashSeparatedCourseKey.from_deprecated_string(kwargs['course_id'])
+            cohort = get_cohort(kwargs['user'], course_key, assign=False)
+            partition_group = None
+            try:
+                partition_group = CourseUserGroupPartitionGroup.objects.get(course_user_group=cohort)
+            except CourseUserGroupPartitionGroup.DoesNotExist:
+                pass
+            if partition_group:
+                partition_group = partition_group.group_id
```

Comment:
@dino-cikatic Strange reassignment of partition_group to its child member. I'd prefer that we do something like this:
`filter_dictionary['content_groups'] = unicode(partition_group.group_id) if partition_group else None`

then add support within edx-search to allow a filter_dictionary specification to be `None`, which would allow us to always filter by a field (like `content_groups`) even when there is no value for that field.

I have taken the liberty of implementing this change in this PR: https://github.com/edx/edx-search/pull/13. Please review and one that is merged we'll want to update the has for edx-search within github.txt